pkg/subctl/cmd/gather/cni.go

```diff
@@ -0,0 +1,161 @@
+}
+
+var networkPluginCNIType = map[string]string{
+	"generic":       typeIPTables,
+	"canal-flannel": typeIPTables,
+	"weave-net":     typeIPTables,
+	"OpenShiftSDN":  typeIPTables,
+	"OVNKubernetes": typeOvn,
+	"unknown":       typeUnknown,
+}
+
+func CNIResources(info Info) {
+	podLabelSelector := RouteagentPodLabel
+	err := func() error {
+		pods, err := findPods(info.ClientSet, podLabelSelector)
+
+		if err != nil {
+			return err
+		}
+
+		info.Status.QueueSuccessMessage(fmt.Sprintf("Found %d pods matching label selector %q", len(pods.Items), podLabelSelector))
+
+		for i := range pods.Items {
+			pod := &pods.Items[i]
+			logIPCmds(info, pod)
+			switch networkPluginCNIType[info.Submariner.Status.NetworkPlugin] {
+			case typeIPTables:
+				logIPTablesCmds(info, pod)
+			case typeOvn:
+				info.Status.QueueWarningMessage("OVN CNI not supported yet")
+			case typeUnknown:
+				info.Status.QueueFailureMessage("Unsupported CNI Type")
+			}
+		}
+		return nil
+	}()
+	if err != nil {
+		info.Status.QueueFailureMessage(fmt.Sprintf("Failed to gather CNI data from pods matching label selector %q: %s",
+			podLabelSelector, err))
+	}
+}
+
+func logIPCmds(info Info, pod *v1.Pod) {
+	for name, cmd := range ipCmds {
+		err := logCmdOutput(info, pod, cmd, name)
+		if err != nil {
+			info.Status.QueueFailureMessage(fmt.Sprintf("%q", err))
+		}
+	}
+}
+
+func logIPTablesCmds(info Info, pod *v1.Pod) {
+	for name, cmd := range ipTablesCmds {
+		err := logCmdOutput(info, pod, cmd, name)
+		if err != nil {
+			info.Status.QueueFailureMessage(fmt.Sprintf("%q", err))
+		}
+	}
+}
+
+func CableDriverResources(info Info) {
+	podLabelSelector := gatewayPodLabel
+	err := func() error {
+		pods, err := findPods(info.ClientSet, podLabelSelector)
+
+		if err != nil {
+			return err
+		}
+
+		info.Status.QueueSuccessMessage(fmt.Sprintf("Found %d pods matching label selector %q", len(pods.Items), podLabelSelector))
+
+		for i := range pods.Items {
+			pod := &pods.Items[i]
+			if info.Submariner.Spec.CableDriver == libreswan {
+				logLibreswanCmds(info, pod)
+			}
+		}
+		return nil
+	}()
+	if err != nil {
+		info.Status.QueueFailureMessage(fmt.Sprintf("Failed to gather CNI data from pods matching label selector %q: %s",
+			podLabelSelector, err))
+	}
+}
+
+func logLibreswanCmds(info Info, pod *v1.Pod) {
+	for name, cmd := range libreswanCmds {
+		err := logCmdOutput(info, pod, cmd, name)
+		if err != nil {
+			info.Status.QueueFailureMessage(fmt.Sprintf("%q", err))
+		}
+	}
+}
+
+func logCmdOutput(info Info, pod *v1.Pod, cmd, cmdName string) error {
+	execOptions := resource.ExecOptionsFromPod(pod)
+	execConfig := resource.ExecConfig{
+		RestConfig: info.RestConfig,
+		ClientSet:  info.ClientSet,
+	}
+	execOptions.Command = []string{"/bin/bash", "-c", cmd}
+	stdOut, _, err := resource.ExecWithOptions(execConfig, execOptions)
+	if err != nil {
+		return errors.WithMessagef(err, "error running %q on pod %s : %q", cmd, pod.Name, err)
```

Comment:
I think we should just log the failure message here as we have all the context and not return error.

```suggestion
               info.Status.QueueFailureMessage(fmt.Sprintf("Error running %q on pod %q: %v", cmd, pod.Name, err))
               return
```